lib/target.py

```diff
@@ -51,11 +51,99 @@ class ArchType:
+# See apple/swift PR #608    
+    @staticmethod
+    def to_string(value):
+        if value == ArchType.arm:
+            return "armv7"
+        if value == ArchType.armeb:
+            return "armeb"
+        if value == ArchType.aarch64:
+            return "aarch64"
+        if value == ArchType.aarch64_be:
+            return "aarch64_be"
+        if value == ArchType.bpfel:
+            return "bpfel"
+        if value == ArchType.bpfeb:
+            return "bpfeb"
+        if value == ArchType.hexagon:
+            return "hexagon"
+        if value == ArchType.mips:
+            return "mips"
+        if value == ArchType.mipsel:
+            return "mipsel"
+        if value == ArchType.mips64:
+            return "mips64"
+        if value == ArchType.mips64el:
+            return "mips64el"
+        if value == ArchType.msp430:
+            return "msp430"
+        if value == ArchType.ppc:
+            return "ppc"
+        if value == ArchType.ppc64:
+            return "ppc64"
+        if value == ArchType.ppc64le:
+            return "ppc64le"
+        if value == ArchType.r600:
+            return "r600"
+        if value == ArchType.amdgcn:
+            return "amdgcn"
+        if value == ArchType.sparc:
+            return "sparc"
+        if value == ArchType.sparcv9:
+            return "sparcv9"
+        if value == ArchType.sparcel:
+            return "sparcel"
+        if value == ArchType.systemz:
+            return "systemz"
+        if value == ArchType.tce:
+            return "tce"
+        if value == ArchType.thumb:
+            return "armv7"
+        if value == ArchType.thumbeb:
+            return "thumbeb"
+        if value == ArchType.x86:
+            return "i386"
+        if value == ArchType.x86_64:
+            return "x86_64"
+        if value == ArchType.xcore:
+            return "xcore"
+        if value == ArchType.nvptx:
+            return "nvptx"
+        if value == ArchType.nvptx64:
+            return "nvptx64"
+        if value == ArchType.le32:
+            return "le32"
+        if value == ArchType.le64:
+            return "le64"
+        if value == ArchType.amdil:
+            return "amdil"
+        if value == ArchType.amdil64:
+            return "amdil64"
+        if value == ArchType.hsail:
+            return "hsail"
+        if value == ArchType.hsail64:
+            return "hsail64"
+        if value == ArchType.spir:
+            return "spir"
+        if value == ArchType.spir64:
+            return "spir64"
+        if value == ArchType.kalimba:
+            return "kalimba"
+        if value == ArchType.shave:
+            return "shave"
+        return "unknown"
+# Not 1:1, See to_string
     @staticmethod
     def from_string(string):
         if string == "arm":
             return ArchType.arm
+        if string == "armv7":
+            return ArchType.arm
+        if string == "armv7l":
+            return ArchType.arm
```

Comment:
i have previously used armv5 armv5tejl armv6 and armv7te; so perhaps this should actually just check a prefix